tools/ci_build/build.py

```diff
@@ -728,6 +730,8 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
             "ON" if args.enable_training else "OFF"),
         "-Donnxruntime_USE_HOROVOD=" + (
             "ON" if args.use_horovod else "OFF"),
+        "-Donnxruntime_USE_NCCL=" + (
```

Comment:
IIUC the intention of this flag was to prevent a build failure when enabling training without CUDA support:
https://github.com/microsoft/onnxruntime/pull/5516
(After the addition of ROCm the build failure changed into a cmake warning).
With the PR above being merge the flag above is likely not needed anymore.
(I am commenting on this PR because @jupvfranco and I are using this change for testing)

Reply:
@alberto-magni Assuming this is not blocking merging this PR, I'd like us to kick off a separate discussion with @weixingzhang @Tixxx and others on the correct set of flags we want, and then make sure we implement them consistently across build.py and the generate makefiles.  We have the Horovod-deprecation PR also to merge, in addition to the progress on the AMD work. 

At the very least we want to be able to build ORT training without any support for distribution, and to build ORT with support for distribution on CPU-only machines, and to build with NCCL.  I am not yet clear on whether we need to break out NCCL vs RCCL.  

I don't think we are consistent yet on which flags must be explicit, and which flags are derived in part from the libraries that cmake finds installed on the build machine.

Reply:
I see, thank you @tlh20. Of course no problems with merging the PR.
I just wanted to highlight the fact that this flag is not needed anymore to prevent a build break.